Explorer/Assets/DCL/CharacterTriggerArea/Tests/CharacterTriggerAreaHandlerSystemShould.cs

```diff
@@ -0,0 +1,504 @@
+
+            Assert.IsFalse(exitTriggerCalled);
+            Assert.IsFalse(enterTriggerCalled);
+
+            system.Update(0);
+            await WaitForPhysics();
+
+            Assert.IsFalse(exitTriggerCalled);
+            Assert.IsFalse(enterTriggerCalled);
+
+            mainPlayerReferences.MainPlayerAvatarBase.SetAvatarBase(fakeMainPlayerAvatarGO.GetComponent<AvatarBase>());
+
+            system.Update(0);
+            await WaitForPhysics();
+
+            Assert.IsTrue(enterTriggerCalled);
+            Assert.IsFalse(exitTriggerCalled);
+        }
+
+        [Test]
+        [TestCase(true)]
+        [TestCase(false)]
+        public async Task DiscriminateCharacterTypeCorrectly(bool onlyMainPlayer)
+        {
+            // Workaround for Unity bug not awaiting async Setup correctly
+            await UniTask.WaitUntil(() => system != null);
+
+            GameObject? fakeOtherPlayerGO = await Addressables.LoadAssetAsync<GameObject>("Character Object");
+            fakeOtherPlayerGO = Object.Instantiate(fakeOtherPlayerGO.GetComponent<CharacterObject>()).gameObject;
+            fakeOtherPlayerGO.transform.position = Vector3.zero;
+
+            var enterTriggerCalls = 0;
+            var exitTriggerCalls = 0;
+
+            var pbComponent = new PBCameraModeArea();
+
+            var component = new CharacterTriggerAreaComponent
+            {
+                AreaSize = Vector3.one * 4,
+                OnEnteredTrigger = collider => enterTriggerCalls++,
+                OnExitedTrigger = collider => exitTriggerCalls++,
+                TargetOnlyMainPlayer = onlyMainPlayer,
+                IsDirty = true,
+            };
+
+            world.Add(entity, component, pbComponent);
+
+            system.Update(0);
+
+            Assert.AreEqual(0, enterTriggerCalls);
+            Assert.AreEqual(0, exitTriggerCalls);
+
+            // Move both characters inside area
+            fakeMainPlayerGO.transform.position = entityTransformComponent.Transform.position;
+            fakeOtherPlayerGO.transform.position = entityTransformComponent.Transform.position;
+
+            await WaitForPhysics();
+
+            Assert.AreEqual(0, exitTriggerCalls);
+            Assert.AreEqual(onlyMainPlayer ? 1 : 2, enterTriggerCalls);
+
+            // Move both characters outside area
+            fakeMainPlayerGO.transform.position += Vector3.one * 30;
+            fakeOtherPlayerGO.transform.position += Vector3.one * 30;
+
+            await WaitForPhysics();
+
+            Assert.AreEqual(onlyMainPlayer ? 1 : 2, enterTriggerCalls);
+            Assert.AreEqual(onlyMainPlayer ? 1 : 2, exitTriggerCalls);
+
+            // Cleanup
+            Object.DestroyImmediate(fakeOtherPlayerGO);
+        }
+
+        private async Task WaitForPhysics()
+        {
+            await UniTask.Yield();
+            await UniTask.Yield();
+            await UniTask.Yield();
+            await UniTask.Yield();
+            await UniTask.Yield();
```

Comment:
I ended up changing it to even more frames, it's needed so that the non-deterministic-hardware CI has time to run the physics simulation, otherwise those tests don't always pass